### ISSUE TITLE
Differences between the Docker and the online versions

### ISSUE THREAD

**Issue by zstadler** [maputnik/editor#487]
I'm using the [Maputnik editor Docker image](https://hub.docker.com/r/maputnik/editor) as part of the [openmaptiles/openmaptiles](https://github.com/openmaptiles/openmaptiles) package. I was expecting the Docker image and the online version at [maputnik.github.io](https://maputnik.github.io/editor) to be synchronized. However, the Docker version looks and behaves differently than the online version. In particular, the Docker version has issues that do not exist in the online version.

How can I get the look and behavior of the Online version with the Docker image?

#### Entering negative or non-integers values in the Docker version
In the Docker version, it is impossible to type negative and non-integer values in numeric fields . For example, entering `-0.5` in the "offset" field is blocked as soon as one the "-" key is pressed. 

It is possible to enter such a number using the "home" or arrow keys. For example "5", "Home", ".", "Home", "-"

The online version allows typing "0.5" and "-.5" but would not allow typing "-0.5"

#### Editing the JSON code in the Docker version
In the Docker version, some edits of the JSON code are not possible. For example, changing
```javascript
    "line-dasharray": [
      0.36,
      0.18
    ],
```
to
```javascript
    "line-dasharray": [
      0.36,
      0.18,
      0.18,
      0.18
    ],
```
is blocked as soon as the first "," key is pressed after "0.18".

#### "View" vs. "Mode"
The Docker version has a "View" drop-down
![image](https://user-images.githubusercontent.com/1304610/51887482-b3cdf380-239c-11e9-8af5-26d24b61d1d9.png)

The online versions show these "Mode" buttons
![image](https://user-images.githubusercontent.com/1304610/51887545-e7a91900-239c-11e9-9bf1-b96aaa509502.png)
![image](https://user-images.githubusercontent.com/1304610/51887564-fd1e4300-239c-11e9-87ec-2a24dc2a7028.png)

### Notes
1. The Klokantech Basic style was used as the initial style in all examples.
2. The Docker version is using `http` while the online version uses `https`.
3. The online version at [openmaptiles.org](http://editor.openmaptiles.org) seems to be similar in behavior to the online version at github.io.

**Comment by pathmapper** [comment]
Hi @zstadler ,

thanks, this is indeed confusing, these are the different versions in use:

- https://maputnik.github.io/editor/ is using the [latest release,](https://github.com/maputnik/editor/releases) currently `v1.5.0`

- the [latest Maputnik image on Dockerhub](https://hub.docker.com/r/maputnik/editor/builds) is build with `master` (which is currently 188 commits ahead of `v1.5.0`). I think we should add docker images tagged for the different releases -> https://github.com/maputnik/editor/issues/489

- http://editor.openmaptiles.org/ is a modified version of `v1.1.0`, this seems to be the repo: https://github.com/klokantech/maputnik/tree/editor-openmaptiles-org
It is outdated and sometimes leads to confusion, see https://github.com/openmaptiles/www.openmaptiles.org/issues/28.

> How can I get the look and behavior of the Online version with the Docker image?

There is currently no such image available on docker hub, but you may build an image with `v1.5.0` yourself:

`docker build https://github.com/maputnik/editor.git#v1.5.0`

> Entering negative or non-integers values in the Docker version

Thanks for reporting. We should fix this for the next release -> https://github.com/maputnik/editor/issues/488

> Editing the JSON code in the Docker version

Also thanks for reporting. This seems to be the same issue as reported here: https://github.com/maputnik/editor/issues/484

> "View" vs. "Mode"

This is expected and was introduced in https://github.com/maputnik/editor/pull/392. However, it is not working in some browsers, see https://github.com/maputnik/editor/issues/472.

> The Docker version is using http while the online version uses https.

What do you mean exactly?

**Comment by zstadler** [comment]
Hi @pathmapper,

Thanks for clearing thing up for me.

>> How can I get the look and behavior of the Online version with the Docker image?

> There is currently no such image available on docker hub, but you may build an image with v1.5.0 yourself:
> `docker build https://github.com/maputnik/editor.git#v1.5.0`

This is unfortunate. Other projects that use the Maputnik's Docker image, such as [OpenMapTiles](https://github.com/openmaptiles/openmaptiles/blob/master/Makefile#L145), have no alternative but to use the cutting-edge builds.

Will it be possible to add a `1.5.0` build to the [available Maputnik tags](https://hub.docker.com/r/maputnik/editor/tags)?

>> The Docker version is using http while the online version uses https.

> What do you mean exactly?

I noticed several issues referring to differences between Maputnik installations serving http and https. I was not sure if it is relevant to the differences I've noticed. Just in case, I wrote that the online version has https interface while a docker-based installation uses http.
